controller/status.go

```diff
@@ -285,29 +276,55 @@ var (
 //   sa: systemd active state
 //   ss: systemd sub state
 //
-func AggregateStatus(fc, fd, sa, ss string) (Status, error) {
+func (a Aggregator) AggregateStatus(fc, fd, sa, ss string) (Status, error) {
+	aggregatedStatuses := []Status{}
+
 	for _, statusContext := range StatusIndex {
-		if !matchState(statusContext.FleetCurrent, fc) {
+		if !a.matchState(statusContext.FleetCurrent, fc) {
 			continue
 		}
-		if !matchState(statusContext.FleetDesired, fd) {
+		if !a.matchState(statusContext.FleetDesired, fd) {
 			continue
 		}
-		if !matchState(statusContext.SystemdActive, sa) {
+		if !a.matchState(statusContext.SystemdActive, sa) {
 			continue
 		}
-		if !matchState(statusContext.SystemdSub, ss) {
+		if !a.matchState(statusContext.SystemdSub, ss) {
 			continue
 		}
 
-		// All requirements matched, so return the aggregated status.
-		return statusContext.Aggregated, nil
+		aggregatedStatuses = append(aggregatedStatuses, statusContext.Aggregated)
+	}
+
+	numAggregatedStatuses := len(aggregatedStatuses)
+
+	if numAggregatedStatuses == 0 {
+		return "", maskAnyf(invalidUnitStatusError, "fc: %s, fd: %s, sa: %s, ss: %s", fc, fd, sa, ss)
+	}
+
+	if numAggregatedStatuses > 1 {
+		a.Logger.Warning(nil, "Multiple statuses found for: fc: %s, fd: %s, sa: %s, ss: %s", fc, fd, sa, ss)
```

Comment:
Do we know when this happens? Neither the issue nor the PR provide some helpful information.